target/linux/ar71xx/image/generic-senao.mk

```diff
@@ -39,3 +39,16 @@ define Device/koala
   IMAGE/sysupgrade.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
 endef
 TARGET_DEVICES += koala
+
+define Device/ursus
+  DEVICE_TITLE := OCEDO Ursus
+  BOARDNAME := URSUS
+  DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca988x
+  KERNEL_SIZE := 1536k
+  IMAGE_SIZE := 7424k
+  MTDPARTS := spi0.0:256k(u-boot)ro,64k(u-boot-env),1536k(kernel),5888k(rootfs),1536k(kernel2),5888k(rootfs2),1088k(data)ro,64k(id)ro,64k(art)ro,7424k@0x50000(firmware)
```

Comment:
Since you are not using the dual image layout, you can use kernel + rootfs + kernel2 + rootfs2 for the firmware. incorporating my other comments, the image build code should look like:

```
define Device/ursus
  DEVICE_TITLE := OCEDO Ursus
  BOARDNAME := URSUS
  IMAGE_SIZE := 14848k
  MTDPARTS := spi0.0:256k(u-boot)ro,64k(u-boot-env),14848k(firmware),1088k(data)ro,64k(id)ro,64k(art)ro
  DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca988x
  IMAGE/sysupgrade.bin := append-kernel | append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
endef
TARGET_DEVICES += ursus
```

Reply:
Please test the resulting image before updating the PR.

Reply:
The flip of the active partition can be automated by doing something similar to https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/base-files/lib/upgrade/platform.sh#L762. It need to be done in `platform_pre_upgrade()`

Reply:
The dual-image is something i'm not sure of. If i'm not mistaken, most devices using this feature set the flag on kernel panics, which we are not able to do.

I'm quite sure, above solution won't work, as uboot might load the kernel from the second partition but the kernel will still mount squashfs from first partition. I've seen this with recent unifi devices.

In my opinion, we should keep the second partition and be it for keeping SteelOS aside (which i think is also a nice feature.